src/Groups/GAPGroups.jl

```diff
@@ -0,0 +1,617 @@
+div_left, div_right!, div_left!, deg, mul, mul!, listperm, degree, elements, right_coset, coset_decomposition,
+right_cosets , right_transversal, conjugacy_class, conjugacy_classes, number_conjugacy_classes, isconjugate, conjugacy_classes_subgroups, conjugacy_classes_maximal_subgroups, normalizer, normaliser, core, pcore, fitting_subgroup, frattini_subgroup, radical_subgroup, socle, sylow_subgroup, hall_subgroup, sylow_system, complement_system, hall_system, relators, isperfect, issimple, isalmostsimple, ispgroup
+
+
+# TODO: as soon as GAP packages like `polycyclic` or `rcwa` are loaded,
+# the custom group types and isos they define should be added to the arrays
+# _gap_group_types resp. _iso_function
+
+
+function group_element(G::T, x::GapObj) where T <: Group
+  return GAPGroupElem{T}(G, x)
+end
+
+function elements(G::T) where T <: Group
+  els = GAP.gap_to_julia(GAP.Globals.Elements(G.X))
+  elems = Vector{elem_type(G)}(undef, length(els))
+  i = 1
+  for x in els
+    elems[i] = group_element(G, x)
+    i += 1
+  end
+  return elems
+end
+
+# to be fixed later
+Base.hash(x::GAPGroupElem) = 0
+
+function parent(x::GAPGroupElem)
+  return x.parent
+end
+
+function Base.isfinite(G::PermGroup)
+  return true
+end
+
+function Base.isfinite(G::PcGroup)
+  return true
+end
+
+function Base.isfinite(G::Group)
+  return GAP.Globals.IsFinite(G.X)
+end
+
+function degree(x::PermGroup)
+   return x.deg
+end
+
+function order(x::Group)
+   return GAP.gap_to_julia(GAP.Globals.Size(x.X))
+end
+
+function order(x::GAPGroupElem)
+   return GAP.gap_to_julia(GAP.Globals.Order(x.X))
+end
+
+function order(::Type{T}, x::Union{GAPGroupElem, Group}) where T<:Number
+   return T(order(x))
+end
+
+Base.exponent(x::Group) = GAP.Globals.Exponent(x.X)
+
+Base.length(x::Group) = order(x)
+
+function Base.rand(x::Group)
+   s = GAP.Globals.Random(x.X)
+   return group_element(x, s)
+end
+
+# one of the following should be non-parametric
+rand_pseudo(G::Group) = Base.rand(G)
+
+
+function _maxgroup(x::T, y::T) where T <: Group
+   if x == y
+     return x
+   elseif GAP.Globals.IsSubset(x.X, y.X)
+     return x
+   elseif GAP.Globals.IsSubset(y.X, x.X)
+     return y
+   else
+     error("Not yet implemented")
+   end
+end
+
+#We need a lattice of groups to implement this properly
+function _prod(x::T, y::T) where T <: GAPGroupElem
+  G = _maxgroup(parent(x), parent(y))
+  return group_element(G, x.X*y.X)
+end
+
+Base.:*(x::GAPGroupElem, y::GAPGroupElem) = _prod(x, y)
+
+function ==(x::Group, y::Group)
+   return x.X == y.X
+end
+
+function ==(x::T, y::T) where T <: GAPGroupElem
+   return x.X == y.X
+end
+
+"""
+    one(G::Group) -> x::GAPGroupElem{typeof(G)}
+Returns the one element of the group G.
+"""
+Base.one(x::Group) = group_element(x, GAP.Globals.Identity(x.X))
+
+"""
+    one(x::GAPGroupElement{T}) -> x::GAPGroupElem{T}
+Returns the one element of the parent of x.
+"""
+Base.one(x::GAPGroupElem) = one(parent(x))
```

Comment:
I think that the intention of all of those methods (which do not fall back on GAP specifics) was to be included into AA and defined for abstract `AbstractAlgebra.Group` and `AbstractAlgebra.GroupElem`

Why not introduce them there and just fall back on those here? @fingolfin your opinion?

Reply:
I'd be happy to do that eventually, but for now we need to get this code working. This is already quite some work; if we have to in parallel also update AbstractAlgebra, and worry about compatibility and releases there, etc., that is too much. Of course this would be a non-issue in a monorepo (luckily the code required for that is now merged into Pkg.jl, so in Julia 1.5 this will be fully supported)

Reply:
Does the syntax of the function work for every `AbstractAlgebra` object?
I mean, if I define the function as

> Base.one(x::AbstractAlgebra.GroupElem) = one(parent(x)),

does it work?

Reply:
Something like this should work, yes -- grepping through AbstractAlgebra, I already see this:
```
src/NCRings.jl:112:one(x::NCRingElem) = one(parent(x))
```

As @kalmarek pointed out, this would then ideally be in AbstractAlgebra.jl. But to do that, we need to figure out where there it perhaps should be, and how to fit in with that; and that's non-trivial, and some of the design already there is totally at odds with how things work in GAP. I'll try to discuss this with @fieker soon. But for now, I'd not worry about modifying AbstractAlgebra.jl; we first need a plan on how to move forward with that.